sr_robot_launch/scripts/sr_ur_arm_unlock

```diff
@@ -0,0 +1,166 @@
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+from std_msgs.msg import Bool
+from ur_dashboard_msgs.srv import GetSafetyMode, GetProgramState, GetRobotMode, Load
+from ur_dashboard_msgs.msg import SafetyMode, ProgramState, RobotMode
+from std_srvs.srv import Trigger
+
+class SrUrUnlock():
+    def __init__(self):
+        self.external_control_program_name = rospy.get_param("~URCap_program_name", "external_ctrl.urp")
+        rospy.loginfo("name: %s", self.external_control_program_name)
+        # self.external_control_program_name = "external_ctrl.urp"
+        self.arms = []
+        if rospy.has_param('ra_sr_ur_robot_hw'):
+            self.arms.append('ra')
+        if rospy.has_param('la_sr_ur_robot_hw'):
+            self.arms.append('la')
+        if len(self.arms) == 0:
+            rospy.logerr("No arms detected, shutting down %s", rospy.get_name())
+            rospy.signal_shutdown("No arms detected")
+        self.subscriber = rospy.Subscriber("/sr_arm/release", Bool, self.pedal_sub)
+        self.rate = rospy.Rate(5)
+        rospy.spin()
+
+    def pedal_sub(self, data):
+        if True == data.data:
+            self.release_arm()
+
+    def call_arm_service(self, side, service_name, service_type, dashboard=True):
+        if dashboard:
+            service_string = "/" + side + "_sr_ur_robot_hw/dashboard/" + service_name
+        else:
+            service_string = "/" + side + "_sr_ur_robot_hw/" + service_name
+        try:
+            service_call = rospy.ServiceProxy(service_string, service_type)
+            response = service_call()
+            return response
+        except rospy.ServiceException, e:
+            rospy.logerr("Service call to '%s' failed for arm %s. %s", service_name, arm, e)
+
+    def startup_arms(self, arms):
+        for arm in arms:
+            self.call_arm_service(arm, "power_on", Trigger)
+        rospy.sleep(5)
+        for arm in arms:
+            self.call_arm_service(arm, "brake_release", Trigger)
+        rospy.sleep(5)
+
+    def check_arms_e_stops(self, arms):
+        for arm in arms:
+            safety_mode = self.call_arm_service(arm, "get_safety_mode", GetSafetyMode)
+            if safety_mode.safety_mode.mode == SafetyMode.ROBOT_EMERGENCY_STOP:
+                while safety_mode.safety_mode.mode == SafetyMode.ROBOT_EMERGENCY_STOP:
+                    rospy.logwarn("Emergency stop button is still pressed for arm %s, please release", arm)
+                    safety_mode = self.call_arm_service(arm, "get_safety_mode", GetSafetyMode)
+                    rospy.sleep(2)
+
+    def check_fix_arms_protective_stop(self, arms):
```

Comment:
lets rename it to something like "unlock_arm_if_protective_stop"

Reply:
Man I had some stupid method names :P 11c9f4658b1c2977eef546c8fbc1abe42126630c